### PR TITLE
en: improve TLS docs

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added, or deleted? (Required)

Add config description for TLS clusters.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/1757
- Other reference link(s):

### CODE REVIEW

File: en/enable-tls-for-mysql-client.md [hunk 1]
```diff
@@ -23,6 +23,14 @@ To enable TLS for the MySQL client, perform the following steps:
 
 2. [Deploy the cluster](#deploy-the-tidb-cluster), and set `.spec.tidb.tlsClient.enabled` to `true`.
 
+    * To skip TLS authentication for internal components as the MySQL client (such as TidbInitializer, Dashboard, Backup, and Restore), you can add the `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` annotation to the cluster's corresponding `TidbCluster`.
```
Comment (qiancai) [inline en/enable-tls-for-mysql-client.md:26]:
```suggestion
    * To skip TLS authentication for internal components that serve as the MySQL client (such as TidbInitializer, Dashboard, Backup, and Restore), you can add the `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` annotation to the cluster's corresponding `TidbCluster`.
```

Comment (qiancai) [reply en/enable-tls-for-mysql-client.md:26]: Perhaps, the updates are inaccurate, but "internal components as the MySQL client" looks confusing. If necessary, please rephrase further.

File: en/enable-tls-for-mysql-client.md [hunk 2]
```diff
@@ -23,6 +23,14 @@ To enable TLS for the MySQL client, perform the following steps:
...
+    * To disable the client CA certificate authentication on the TiDB server, you can set `.spec.tidb.tlsClient.disableClientAuthn` to `true`. This means skipping setting `ssl-ca` parameter when you [configure TiDB server to enable secure connections](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections).
```
Comment (qiancai) [inline en/enable-tls-for-mysql-client.md:27]:
```suggestion
    * To disable the client CA certificate authentication on the TiDB server, you can set `.spec.tidb.tlsClient.disableClientAuthn` to `true`. This means skipping setting the `ssl-ca` parameter when you [configure TiDB server to enable secure connections](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections).
```

File: en/enable-tls-for-mysql-client.md [hunk 3]
```diff
@@ -23,6 +23,14 @@ To enable TLS for the MySQL client, perform the following steps:
...
+    * To skip the CA certificate authentication for internal components as the MySQL client, you can set `.spec.tidb.tlsClient.skipInternalClientCA` to `true`.
```
Comment (qiancai) [inline en/enable-tls-for-mysql-client.md:28]: ditto

File: en/enable-tls-for-mysql-client.md [hunk 4]
```diff
@@ -23,6 +23,14 @@ To enable TLS for the MySQL client, perform the following steps:
...
+
+    > **Note:**
+    >
+    > For an existing cluster, if you change `.spec.tidb.tlsClient.enabled` from `false` to `true`, the TiDB Pod will be rolling restarted.
```
Comment (qiancai) [inline en/enable-tls-for-mysql-client.md:32]:
```suggestion
    > For an existing cluster, if you change `.spec.tidb.tlsClient.enabled` from `false` to `true`, the TiDB Pods will be rolling restarted.
```

**Review by qiancai — COMMENTED** [review]
@ran-huang I've completed the review and added some comments. PTAL. Thanks.

**Review by just1900 — APPROVED** [review]

File: en/enable-tls-for-mysql-client.md (hunk 1 above)
Comment (ran-huang) [reply en/enable-tls-for-mysql-client.md:26]: I think your suggestion looks better. Thanks.

File: en/enable-tls-for-mysql-client.md [hunk 5]
```diff
@@ -23,6 +23,14 @@ To enable TLS for the MySQL client, perform the following steps:
 
 2. [Deploy the cluster](#deploy-the-tidb-cluster), and set `.spec.tidb.tlsClient.enabled` to `true`.
 
+    * To skip TLS authentication for internal components that serve as the MySQL client (such as TidbInitializer, Dashboard, Backup, and Restore), you can add the `tidb.tidb.pingcap.com/skip-tls-when-connect-tidb="true"` annotation to the cluster's corresponding `TidbCluster`.
+    * To disable the client CA certificate authentication on the TiDB server, you can set `.spec.tidb.tlsClient.disableClientAuthn` to `true`. This means skipping setting the `ssl-ca` parameter when you [configure TiDB server to enable secure connections](https://docs.pingcap.com/tidb/stable/enable-tls-between-clients-and-servers#configure-tidb-server-to-use-secure-connections).
+    * To skip the CA certificate authentication for internal components as the MySQL client, you can set `.spec.tidb.tlsClient.skipInternalClientCA` to `true`.
```
Comment (ran-huang) [inline en/enable-tls-for-mysql-client.md:28]:
```suggestion
    * To skip the CA certificate authentication for internal components that serve as the MySQL client, you can set `.spec.tidb.tlsClient.skipInternalClientCA` to `true`.
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**